app/org/sagebionetworks/bridge/play/controllers/AuthenticationController.java

```diff
@@ -42,8 +42,10 @@ public Result signOut() throws Exception {
 
     public Result signUp() throws Exception {
         JsonNode json = requestToJSON(request());
-        SignUp signUp = SignUp.fromJson(json, false);
-        signUp.getRoles().clear();
+        SignUp signUp = parseJson(request(), SignUp.class);
+        // this is now the only way to eliminate roles if set by the user...
+        signUp = new SignUp(signUp.getUsername(), signUp.getEmail(), 
+                signUp.getPassword(), null, signUp.getDataGroups()); 
```

Comment:
Do we have a test to make sure user's can't sign up with a role? (This could be a security hole if they can specify "admin", so we should have explicit tests for this.)